soroban-env-host/src/host/conversion.rs

```diff
@@ -333,4 +335,12 @@ impl Host {
             }
         }
     }
+
+    pub(crate) fn deserialize_asset(&self, asset: Object) -> Result<Asset, HostError> {
+        self.visit_obj(asset, |hv: &Vec<u8>| {
+            self.charge_budget(CostType::ValDeser, hv.len() as u64)?;
+            Asset::from_xdr(&mut hv.as_slice())
```

Comment:
Can you use the `metered_from_xdr` function instead?
https://github.com/stellar/rs-soroban-env/blob/main/soroban-env-host/src/host/metered_xdr.rs#L50

Reply:
That's nice, I've also added an `Object` helper to avoid `visit_obj` boilerplate in callers.